src/goto-programs/goto_program.h

```diff
@@ -408,13 +408,14 @@ class goto_programt
     {
     }
 
-    explicit instructiont(goto_program_instruction_typet _type):
-      source_location(static_cast<const source_locationt &>(get_nil_irep())),
-      type(_type),
-      guard(true_exprt()),
-      location_number(0),
-      loop_number(0),
-      target_number(nil_target)
+    explicit instructiont(goto_program_instruction_typet _type)
+      : code(ID_nil),
+        source_location(static_cast<const source_locationt &>(get_nil_irep())),
```

Comment:
This change adds non-trivial cost when constructing `instructiont` -- I am wondering whether we should instead move forward on the consolidation of the ireps in there.

Reply:
How about `code(static_cast<codet &>(get_nil_irep())` as a cheaper alternative? Of course the proper consolidation is the correct long-term solution.

Reply:
Done.